packages/typescript-estree/src/convert.ts

```diff
@@ -1841,7 +1838,7 @@ export class Converter {
           value: '',
         });
         result.raw = this.ast.text.slice(result.range[0], result.range[1]);
-        if ((parent as any).name && (parent as any).name === node) {
+        if ('name' in parent && parent.name === node) {
```

Comment:
new condition is logically same, because we do not need to check `if node === null/undefined `